core/desugarModules.ml

```diff
@@ -307,16 +307,28 @@ and desugar ?(toplevel=false) (renamer' : Epithet.t) (scope' : Scope.t) =
         raise (Errors.module_error ~pos (Printf.sprintf "Unbound module %s" (Scope.Resolve.best_guess path)))
 
     method! funlit : funlit -> funlit
-      = fun (paramss, body) ->
-      let visitor = self#clone in
-      let paramss' =
-        List.map
-          (fun params ->
-            List.map (fun param -> visitor#pattern param) params)
-          paramss
-      in
-      let body' = visitor#phrase body in
-      (paramss', body')
+      = fun f ->
+      match f with
+        | NormalFunlit (paramss, body) -> 
+          let visitor = self#clone in
+          let paramss' =
+            List.map
+              (fun params ->
+                List.map (fun param -> visitor#pattern param) params)
+              paramss
+          in
+          let body' = visitor#phrase body in
+          NormalFunlit (paramss', body')
+        | MatchFunlit (paramss, body) -> 
```

Comment:
If I understand correctly, `MatchFunLit` will have been desugared away before module desugaring starts.  So this case can just be `assert false`.